### PR TITLE
Add support for remote growl server

### PR DESCRIPTION
**Pull request by kyriesent**
`checkGrowl` will fail if trying to run against a remote growl server by setting growl host.

Using the example from the docs:

```js
var Growl = require('node-notifier').Growl;

var GROWL_HOST = '10.0.2.2';

var notifier = new Growl({
  name: 'Growl Name Used', // Defaults as 'Node'
  host: GROWL_HOST,
  port: 23053
});

notifier.notify({
  title: 'Foo',
  message: 'Hello World',
  wait: false, // if wait for user interaction

  // and other growl options like sticky etc.
  sticky: false,
  label: void 0,
  priority: void 0
}, function (err, res) {
  console.log(err, res);
});
```

This gives an error that growl can't be connected to. `checkGrowl` is trying to connect to 'localhost', which fails.

### CODE REVIEW

**Comment by kurisubrooks** [comment]
Could you please check as to why the build has failed on this commit? Might be an error within your code.

**Comment by kyriesent** [comment]
@kurisubrooks thanks for the note. I didn't account for growl clients that are left at default configuration, and the tests passed on 0.10 which I was running. It's green now!

If there's anything else I can do, let me know. Thanks for reviewing!! :bow:

**Comment by mikaelbr** [comment]
Hi and thanks for the PR! Could you also fix this for the baloon notifier. It also uses `checkGrowl` for fallback.

**Comment by kyriesent** [comment]
Hi @mikaelbr! Thanks for taking it!

I'd be more than happy to, but can you help me understand the fallback for `balloon`? Let me know if this is right:

1. If growl is installed on 'localhost:23053', then use growl
2. Otherwise, use balloon notifier

This PR should follow this behavior, as checkGrowl will default to checking 'localhost:23053'.

It seems like the balloon notifier doesn't take options for `host` or `port`, so I kept checkGrowl looking for the default.

Is this right? Or is there an option for `balloon` that I should implement that I'm missing?

**Comment by mikaelbr** [comment]
How balloon works is like this:
1. Is fallback activated and the system is Windows 8 or higher? Use Toaster instead (better design/features)
2. Is fallback activated and we detect there is growl? Use Growl instead (better design/features)

The port/host options should be passed through the constructor just as with Growl (here: https://github.com/mikaelbr/node-notifier/blob/master/notifiers/balloon.js#L40) as we in [index pass in the same options](https://github.com/mikaelbr/node-notifier/blob/master/index.js#L13-L34). Unless they explicitly try Balloon constructor. In that case they are required to pass inn options them self.

So in short, you just have to fix the usage of `checkGrowl` in Balloon in the same way you already have in Growl. Toaster uses Balloon as fallback and is implicitly covered by the same code (if that makes any sense to you)

**Comment by kyriesent** [comment]
Got it, thank you!! I will do that right away.

**Comment by mikaelbr** [comment]
Great. Thanks for your work! :cake:

**Comment by kyriesent** [comment]
:raised_hands:
